pkg/lib/cast/interface.go

```diff
@@ -694,6 +694,37 @@ func InterfaceToStrInterfaceMap(in interface{}) (map[string]interface{}, bool) {
 	return out, true
 }
 
+func InterfaceToStrInterfaceMapRecursive(in interface{}) (map[string]interface{}, bool) {
+	if in == nil {
+		return nil, true
+	}
+
+	inMap, ok := InterfaceToInterfaceInterfaceMap(in)
+	if !ok {
+		return nil, false
+	}
+
+	out := map[string]interface{}{}
+
+	for key, value := range inMap {
+		casted, ok := key.(string)
+		if !ok {
+			return nil, false
+		}
+
+		if intToIntMap, ok := value.(map[interface{}]interface{}); ok {
```

Comment:
will this work when the nested map is a map[string]interface{}? If not, you might consider using `InterfaceToInterfaceInterfaceMap()`

Reply:
update: I tried it in go playground ([code](https://play.golang.org/p/7-cYZDUxFHK)), and the cast seems to not work so I think we might want `InterfaceToInterfaceInterfaceMap()` here

Reply:
update2: I see you have a test for this, so I'm a bit confused as to why it passes. We can discuss live